diff -up openjdk/jdk/src/share/classes/sun/security/x509/AlgorithmId.java.sav openjdk/jdk/src/share/classes/sun/security/x509/AlgorithmId.java
--- openjdk/jdk/src/share/classes/sun/security/x509/AlgorithmId.java.sav	2013-01-14 12:04:57.096097896 -0500
+++ openjdk/jdk/src/share/classes/sun/security/x509/AlgorithmId.java	2013-01-14 12:05:14.956835554 -0500
@@ -120,20 +120,8 @@ public class AlgorithmId implements Seri
         try {
             algParams = AlgorithmParameters.getInstance(algidString);
         } catch (NoSuchAlgorithmException e) {
-            try {
-                // Try the internal EC code so that we can fully parse EC
-                // keys even if the provider is not registered.
-                // This code can go away once we have EC in the SUN provider.
-                algParams = AlgorithmParameters.getInstance(algidString,
-                                sun.security.ec.ECKeyFactory.ecInternalProvider);
-            } catch (NoSuchAlgorithmException ee) {
-                /*
-                 * This algorithm parameter type is not supported, so we cannot
-                 * parse the parameters.
-                 */
-                algParams = null;
-                return;
-            }
+            algParams = null;
+            return;
         }
         // Decode (parse) the parameters
         algParams.init(params.toByteArray());
