diff -up openjdk/jdk/src/share/lib/security/java.security.sav openjdk/jdk/src/share/lib/security/java.security
--- openjdk/jdk/src/share/lib/security/java.security.sav	2013-02-01 17:34:10.750939826 -0500
+++ openjdk/jdk/src/share/lib/security/java.security	2013-02-01 17:36:09.283198885 -0500
@@ -148,7 +148,9 @@
                com.sun.org.apache.xml.internal.utils.,\
                com.sun.org.apache.xml.internal.security.,\
                com.sun.org.glassfish.,\
-               org.jcp.xml.dsig.internal.
+               org.jcp.xml.dsig.internal.,\
+               org.GNOME.Accessibility.,\
+               org.GNOME.Bonobo.
 
 #
 # List of comma-separated packages that start with or equal this string
@@ -185,7 +187,9 @@
                    com.sun.org.apache.xml.internal.utils.,\
                    com.sun.org.apache.xml.internal.security.,\
                    com.sun.org.glassfish.,\
-                   org.jcp.xml.dsig.internal.
+                   org.jcp.xml.dsig.internal.,\
+                   org.GNOME.Accessibility.,\
+                   org.GNOME.Bonobo.
 
 #
 # Determines whether this properties file can be appended to
