diff --git a/src/share/classes/java/util/logging/Logger.java b/src/share/classes/java/util/logging/Logger.java
--- a/src/share/classes/java/util/logging/Logger.java
+++ b/src/share/classes/java/util/logging/Logger.java
@@ -307,7 +307,7 @@
         LogManager manager = LogManager.getLogManager();
         SecurityManager sm = System.getSecurityManager();
         if (sm != null && !SystemLoggerHelper.disableCallerCheck) {
-            // 0: Reflection 1: Logger.getLoggerContext 2: Logger.getLogger 3: caller
+            // 0: Reflection 1: Logger.demandLogger 2: Logger.getLogger 3: caller
             final int SKIP_FRAMES = 3;
             Class<?> caller = sun.reflect.Reflection.getCallerClass(SKIP_FRAMES);
             if (caller.getClassLoader() == null) {
@@ -424,7 +424,7 @@
 
         // all loggers in the system context will default to
         // the system logger's resource bundle
-        Logger result = demandLogger(name, null);
+        Logger result = manager.demandSystemLogger(name, SYSTEM_LOGGER_RB_NAME);
         return result;
     }
 
