diff -r d24281f1ddc2 -r ca2cc93a6640 openjdk/hotspot/make/linux/makefiles/zeroshark.make
--- a/openjdk/hotspot/make/linux/makefiles/zeroshark.make	Wed Jul 24 22:26:51 2013 +0100
+++ a/openjdk/hotspot/make/linux/makefiles/zeroshark.make	Thu Jul 25 20:28:04 2013 +0100
@@ -42,7 +42,7 @@
 
 bytecodes_arm.s: bytecodes_arm.def mkbc
 	@echo Generating ARM assembler bytecode sequences
-	$(CXX_COMPILE) -E - < $< | ./mkbc - $@ $(COMPILE_DONE)
+	$(CXX_COMPILE) -E -x c++ - < $< | ./mkbc - $@ $(COMPILE_DONE)
 
 mkbc:	$(GAMMADIR)/tools/mkbc.c
 	@echo Compiling mkbc tool
